Reset  button requests in device authenticity check

komret opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
Without resetting the button requests, one could get to a state where Suite incorrectly expected confirmation on Trezor, see attached videos 1 and 2.

While fixing this, I encountered some other issues such as unnecessary complexity in the components, missing loading state (video 3) and this condition never triggering:

```ts
if (typeof result === 'string' || result.valid === undefined) {
    dispatch(reportToSentry(new Error(`Device authenticity failed: ${result}`)));
    setIsAborted(true);
}
```



## Related Issue

Resolve <!--- link the issue here -->

## Screenshots:
1)
https://github.com/trezor/trezor-suite/assets/42465546/845034c2-95a1-42c1-afa9-bbd06dee46ed

2)
https://github.com/trezor/trezor-suite/assets/42465546/ecd58b2b-72c8-486f-bef8-a1405c75bd33

3)
https://github.com/trezor/trezor-suite/assets/42465546/1c7b645e-cc5b-41b5-b8ca-8d93bf4fc486


I initially refactored `checkDeviceAuthenticityThunk` to always return a value, but changed my mind in the last two commits -  `StoredAuthenticateDeviceResult` is sent to the reducer and the thunk has no return value. This allowed unifying Sentry reporting in one place and removing the need for the `unwrap` method.

matejkriz commented on packages/suite/src/views/onboarding/steps/SecurityCheck/DeviceAuthenticity.tsx:
```diff
@@ -27,33 +24,29 @@ const StyledExplainer = styled(DeviceAuthenticationExplainer)`
+        request => request.code === 'ButtonRequest_Other',
+    );
+    const isCheckFailed = isSubmitted && deviceAuthenticity?.valid === false;
+    const isCheckSuccessFul = isSubmitted && deviceAuthenticity?.valid;
```
typo?
```suggestion
    const isCheckSuccessful = isSubmitted && deviceAuthenticity?.valid;
```

matejkriz reviewed (approved):
Very nice simplification 👏 👍 

And from my testing it works only better than before. 

Here is testing sentry issue for CA_PUBKEY_NOT_FOUND & expired config: https://satoshilabs.sentry.io/issues/4555687017/events/c9829bada33b461487bc086e4db73b65/

I found just a typo? isCheckSuccessful -> isCheckSuccessFul 

And test coverage needs to be resolved one way or another...

komret commented:
I added unit tests for the actions. Sentry logging is not included in the tests due to `sentryMiddleware` being in another package, I guess there is no way to test it in one place. I noticed another bug while writing the tests, success toast was missing in case of `'CA_PUBKEY_NOT_FOUND'`.  I cleaned up `checkDeviceAuthenticityThunk` a bit more.

matejkriz reviewed (approved):
re-approving again two new commits 👍 great that you didn't just lower the coverage limit, but added useful tests 🖤

STew790 commented:
**QA OK**
All 3 cases now works correctly.

**Info:**
 - Suite version: desktop 23.12.1 (9ec6ac5438d7a952abcbbdc049bf729a47d4e039)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.12.1 Chrome/118.0.5993.129 Electron/27.0.4 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: Trezor T2T1 2.6.3 regular (revision 2c7cc6e0255dee2339b445b5551eaffb88dbd1b4)
 - Transport: BridgeTransport 2.0.33
